docs/architecture.md

```diff
@@ -1,33 +1,67 @@
-# MiniMint Architecture
+# Minimint Architecture
 
-MiniMint started out as a federated Chaumian e-cash prototype. By now it is a more general framework for federated financial applications. At its core lies the ability to agree on and process transactions. The possible input and output types of these transactions are defined by modules.
+Minimint is a general framework for building federated financial applications.
+The current implementation allows users to make private low-fee payments through a federated blinded mint that issues [e-cash](https://en.wikipedia.org/wiki/Ecash) and processes deposits and withdrawals over Bitcoin or Lightning.
 
-To implement the federated e-cash functionality there currently exist two modules:
-* **Fediwallet**: a federated on-chain wallet, supporting deposits and withdrawals
-* **Fedimint**: a federated e-cash mint, supporting issuance and spending of blind signed tokens of diiferent denominations
+## Crate organization
+The [Minimint federation](#Federation-Nodes) consists of nodes that are primarily built from the following crates:
+* `minimint` - the main consensus code for processing transactions and REST API
+* `minimint-api` - the common serialization and database representations for communication between crates
+* `minimint-derive` - helper macros for serialization
 
-In the future other modules, e.g. to integrate Lightning, smart contracts or even a federated market place could be implemented.
+[Modules](#Modules) can be added to the `MinimintConsensus` to allow for new types of transactions and federated actions:
+* `modules/minimint-wallet` - an on-chain bitcoin wallet
+* `modules/minimint-mint` - a blinded mint that issues e-cash
+* `modules/minimint-ln` - a Lightning payment service
 
-## Main loop
-The main functionality is implemented in one big loop shown below.
+The [user client](#User-Client):
+* `mint-client` - provides a client for sending transactions to the federation
 
-![Control and data flow in MiniMint](./architecture.svg)
+The [LN gateway](#LN-Gateway):
+* `ln-gateway` - allows a Lightning node operator to receive or pay Lightning invoices on behalf of users
```

Comment:
maybe some day this can be explained more deeply, I could see how people who read this would have many questions about this.

Reply:
I begin that description in the LN Gateway section, but could definitely have more details in a future revision.